src/pudl/etl.py

```diff
@@ -434,8 +448,9 @@ def etl(  # noqa: C901
         sqlite_dfs.update(_etl_ferc1(datasets["ferc1"], pudl_settings))
     if datasets.get("eia", False):
         sqlite_dfs.update(_etl_eia(datasets["eia"], ds_kwargs))
+    logger.info(sqlite_dfs.keys())
     if datasets.get("glue", False):
-        sqlite_dfs.update(_etl_glue(datasets["glue"]))
+        sqlite_dfs.update(_etl_glue(datasets["glue"], ds_kwargs, sqlite_dfs))
```

Comment:
If this only needs the `generators_entity_eia`, I think just passing in that dataframe is better. The pattern we've had of passing around all of the dataframes to everywhere makes dependencies unclear and exposes all of the dataframes to mutation.

Reply:
That's what I was doing before, and it was breaking the tests. Because the `generators_entity_eia` table doesn't exist in the sqlite_dfs dict when the solo ferc tests are run it breaks. When you pass it the sqlite_dfs dict it doesn't break because the dict still exists. The crosswalk just won't get added because it depends on `glue_settings.eia` which indicates whether eia data has also been integrated.